pkg/controller/controller_instance.go

```diff
@@ -157,6 +158,46 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1alpha1.ServiceIn
 		return nil
 	}
 
+	// Determine if any credentials exist for this instance.  We don't want to
+	// delete the instance if there are any associated creds
+	credentialsLister := c.bindingLister.ServiceInstanceCredentials(instance.Namespace)
+
+	selector := labels.NewSelector()
+	credentialsList, err := credentialsLister.List(selector)
+	if err != nil {
+		return err
+	}
+	for _, credentials := range credentialsList {
+		if instance.Name == credentials.Spec.ServiceInstanceRef.Name {
+
+			// found credentials, block the deletion until they are removed
+			clone, err := api.Scheme.DeepCopy(instance)
+			if err != nil {
+				return err
+			}
+			toUpdate := clone.(*v1alpha1.ServiceInstance)
+
+			s := fmt.Sprintf(
+				"Delete instance %v/%v blocked by existing ServiceInstanceCredentials %v/%v.  All credentials must be removed first.",
```

Comment:
I would take out the instance credentials name and just say there are existing credentials.  We don't want to have to maintain the list of credentials here.